main.cpp

```diff
@@ -0,0 +1,228 @@
+// Configuration editor
+// Command line utiltiy to create configuration files
+// Valid arguments are init and edit
+
+#include <iostream>
+#include <string>
+#include <fstream>
+#include <stdio.h>
+
+
+using std::cout;
+using std::endl;
+using std::string;
+using std::cin;
+
+const string CONFIG_FILE_PATH = "/users/bconway/SourceTree/School/CPT-180-27-Assignment-2-Config-Generator/app.cfg";
+
+
+int main(int argc, char *argv[]) {
+
+    string argument1 = "";
+    // Make sure that the argument isn't empty
+    if (argv[1] != nullptr) {
+        argument1 = static_cast<string>(argv[1]);
+    }
+
+    if (argument1 == "init") {
+
+        // initialize the config file
+        // TODO determine if the file exists and handle the exception
+
+        std::ofstream config_file;
+
+        string temp_value = "";
+
+        config_file.open(CONFIG_FILE_PATH);
+
+        cout << "Enter your name (first and last):" << endl;
+        std::getline(cin, temp_value);
+
+        // If nothing was entered then ask again.
+        while (temp_value == "") {
+            cout << "Error, please enter your name (first and last):" << endl;
+            std::getline(cin, temp_value);
+        }
+        config_file << "[name]=" << temp_value << endl;
+        temp_value = "";
```

Comment:
This is a bad idea to reuse this variable.
If you created a fresh string for each block you would not need to worry about the comment because each one would be `std::string temp_name` and include the property being gathered. This would serve to document the section as well.